client/js/table-model.js

```diff
@@ -16,6 +16,21 @@ class TableModel {
   setValue(location, value) {
     this.data[this._getCellId(location)] = value;
   }
+
+  getSum(column) {
+    let sum = [];
+    for (var prop in this.data) {
+      if (parseInt(prop[0], 10) === column) {
```

Comment:
This would be cleaner if you did the processing in distinct stages, separating out a helper function for stage 1.

1) gather all the values from the target column
2) filter to just numeric values
3) if you have any, sum them